### PR TITLE
[nrf fromlist] Add support for the nRF54H20 DK board

### PR DESCRIPTION
**Pull request by anangl**
Commits cherry picked from:
- https://github.com/zephyrproject-rtos/zephyr/pull/69747 (one commit brought as [nrf fromlist] because it has to be modified downstream due to hwmv2)
- https://github.com/zephyrproject-rtos/zephyr/pull/69924
- https://github.com/zephyrproject-rtos/zephyr/pull/69962
- https://github.com/zephyrproject-rtos/zephyr/pull/70062
- https://github.com/zephyrproject-rtos/zephyr/pull/70154
- https://github.com/zephyrproject-rtos/zephyr/pull/70172
- https://github.com/zephyrproject-rtos/zephyr/pull/70205
- https://github.com/zephyrproject-rtos/zephyr/pull/70411
- https://github.com/zephyrproject-rtos/zephyr/pull/70454
- https://github.com/zephyrproject-rtos/zephyr/pull/70491
- https://github.com/zephyrproject-rtos/zephyr/pull/70616

### CODE REVIEW

File: boards/arm/nrf54h20dk_nrf54h20/board.cmake [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: Apache-2.0
+
+include(${ZEPHYR_BASE}/boards/common/nrfjprog.board.cmake)
```
Comment (gmarull) [inline boards/arm/nrf54h20dk_nrf54h20/board.cmake:3]: pending removal

**Review by FrancescoSer — APPROVED** [review]

**Review by SebastianBoe — APPROVED** [review]

File: boards/arm/nrf54h20dk_nrf54h20/board.cmake (hunk 1 above)
Comment (anangl) [reply boards/arm/nrf54h20dk_nrf54h20/board.cmake]: removed

**Review by gmarull — APPROVED** [review]

**Comment by anangl** [comment]
Rebased.

**Comment by anangl** [comment]
Added commits cherry-picked from https://github.com/zephyrproject-rtos/zephyr/pull/70411.
Replaced `[nrf fromlist]` commits with `[nrf fromtree]` where appropriate.

**Comment by anangl** [comment]
The compliance checks fails in the same mysterious way as in https://github.com/zephyrproject-rtos/zephyr/pull/70411:
```
Error: See https://www.pylint.org/ for more details

 R0801:Similar lines in 2 files
File:scripts/west_commands/tests/test_nrf.py
Line:1
Column:0
 R0801:Similar lines in 2 files
File:scripts/west_commands/tests/test_nrf.py
Line:1
Column:0
```

**Comment by gmarull** [comment]
> The compliance checks fails in the same mysterious way as in [zephyrproject-rtos/zephyr#70411](https://github.com/zephyrproject-rtos/zephyr/pull/70411):
> 
> ```
> Error: See https://www.pylint.org/ for more details
> 
>  R0801:Similar lines in 2 files
> File:scripts/west_commands/tests/test_nrf.py
> Line:1
> Column:0
>  R0801:Similar lines in 2 files
> File:scripts/west_commands/tests/test_nrf.py
> Line:1
> Column:0
> ```

please cherry-pick https://github.com/zephyrproject-rtos/zephyr/pull/70454 (and drop last 3 commits)

**Comment by anangl** [comment]
> please cherry-pick [zephyrproject-rtos/zephyr#70454](https://github.com/zephyrproject-rtos/zephyr/pull/70454) (and drop last 3 commits)

done

**Comment by anangl** [comment]
Rebased.

**Comment by anangl** [comment]
Added commits from https://github.com/zephyrproject-rtos/zephyr/pull/70154.

**Comment by anangl** [comment]
Rebased.

**Comment by anangl** [comment]
Rebased.

**Comment by anangl** [comment]
Rebased.

**Pull request merged**